### PR TITLE
Multiple Bolt transactions associated with same order fix

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
After overriding the initial order statuses and split the order processing into initial order creation and hook processing, due to ERP support and checkout speed up, some edge case flows were broken leading to unwanted results. Check the description in Asana task linked below.

Fixes: https://app.asana.com/0/941920570700290/1144813337953035/f

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] I have created or modified unit tests to sufficiently cover my changes.

### CODE REVIEW

**Comment by danac-gs** [comment]
@matt-thomason can we include this in the release?
I expect @OleksiiOnyshchenko to run into some issues while testing without it.

File: Helper/Order.php [hunk 1]
```diff
@@ -794,12 +794,24 @@ public function processExistingOrder($quote, $transaction)
...
+                        $quote->getReservedOrderId(),
+                        $quote->getId()),
                     null,
                     CreateOrder::E_BOLT_REJECTED_ORDER
                 );
             }
 
+            if ($order->getState() === OrderModel::STATE_PENDING_PAYMENT) {
+                throw new BoltException(
+                    __('Order is in the initial state. Waiting for the hook update. Order #: %1 Quote ID: %2',
```
Comment (matt-thomason) [inline Helper/Order.php]: I think it's clearer to have "Order is in pending payment"

Comment (danac-gs) [reply Helper/Order.php]: done

**Comment by danac-gs** [comment]
@matt-thomason, tests are included. Merging into master.

**Pull request merged**